prov/rxm/src/rxm_ep.c

```diff
@@ -2417,12 +2470,17 @@ int rxm_endpoint(struct fid_domain *domain, struct fi_info *info,
 		goto err1;
 	}
 
+	coll_requested = rxm_ep->rxm_info->caps & FI_COLLECTIVE;
+
 	if (fi_param_get_int(&rxm_prov, "comp_per_progress",
 			     (int *)&rxm_ep->comp_per_progress))
 		rxm_ep->comp_per_progress = 1;
 
+	rxm_ep->txrx_ops = (coll_requested) ? &rxm_coll_rx_ops : &rxm_rx_ops;
+
+	progress = (coll_requested) ? &rxm_ep_progress_coll : &rxm_ep_progress;
```

Comment:
Remove coll_requested int and combine settings into single if-else statement. e.g.
```
if (rxm_ep->rxm_info->caps & FI_COLLECTIVE)
    blah blah
else
    blah blah
```
removes stack variable and extra branch